src/btree/bt_page.c

```diff
@@ -513,6 +763,30 @@ __page_reconstruct_leaf_delta(WT_SESSION_IMPL *session, WT_REF *ref, WT_ITEM *de
     return (ret);
 }
 
+/*
+ * __wti_build_full_disk_image_on_read --
+ *     Build a full disk image of the page after reading from disk.
+ */
+int
+__wti_build_full_disk_image_on_read(
+  WT_SESSION_IMPL *session, WT_REF *ref, WT_ITEM *deltas, size_t delta_size, WT_ITEM *disk_image)
+{
+    WT_DECL_RET;
+    WT_REF **refs;
+    size_t refs_entries, incr;
+
+    refs = NULL;
+    refs_entries = 0;
+    incr = 0;
+
+    /* Merge deltas directly with the base image to build refs in a single pass. */
+    WT_ERR(__page_merge_deltas_with_base_image_new(
+      session, ref, deltas, delta_size, &refs, &refs_entries, &incr, disk_image));
+
+err:
```

Comment:
no need for this

Reply:
Fixed in d6dea8b67ab5a839fbaea965dfe6c4d3190c9d4e